src/ceph_osd.cc

```diff
@@ -117,6 +117,19 @@ static void usage()
   generic_server_usage();
 }
 
+entity_addrvec_t make_osd_addrs(entity_addr_t a)
+{
+  entity_addrvec_t addrs;
+  if (a.get_port() == 0) {
+    a.set_type(entity_addr_t::TYPE_MSGR2);
+    a.set_port(6800);
```

Comment:
Nah, an OSD doesn't bind to a specific, frozen port; in such cases the messenger typically goes with `ms_bind_port_min` and `ms_bind_port_max`.